src/@components/CardCollectionPage/CardSlider/index.tsx

```diff
@@ -1,49 +1,44 @@
-import "slick-carousel/slick/slick.css";
-import "slick-carousel/slick/slick-theme.css";
+import "swiper/swiper.css";
 
 import { forwardRef, useRef } from "react";
-import Slider from "react-slick";
 import { useRecoilState } from "recoil";
+import { Swiper, SwiperRef, SwiperSlide } from "swiper/react";
 
 import { sliderIdxState } from "../../../core/atom/slider";
 import { CardList } from "../../../types/cardCollection";
 import Card from "../Card";
 import LastCard from "../Card/LastCard";
-import { St } from "./style";
-
-const sliderSettings = {
-  className: "center",
-  centerMode: true,
-  arrows: false,
-  dots: false,
-  infinite: false,
-  variableWidth: true,
-  slidesToScroll: 1,
-};
-
+import useCardSwiper from "../hooks/useCardSwiper";
+import St from "./style";
 interface CardSliderProps {
   openLoginModalHandler: () => void;
   cardLists: CardList[];
 }
 
 const CardSlider = forwardRef(function CardSlider(props: CardSliderProps, ref: React.ForwardedRef<HTMLDivElement>) {
   const { openLoginModalHandler, cardLists } = props;
+  const { swiperSettings } = useCardSwiper();
 
   const [sliderIdx, setSliderIdx] = useRecoilState(sliderIdxState);
-  const sliderRef = useRef<Slider | null>(null);
+  const swiperRef = useRef<SwiperRef | null>(null);
 
   return (
     <St.Wrapper>
-      <Slider
-        {...sliderSettings}
+      <Swiper
+        {...swiperSettings}
+        className="swiper"
         initialSlide={sliderIdx}
-        afterChange={(idx: number) => setSliderIdx(idx)}
-        ref={sliderRef}>
+        onSlideChange={(swiper) => setSliderIdx(swiper.activeIndex)}
+        ref={swiperRef}>
```

Comment:
className은 우리가 추가한 prop이지만
initialSlide, onSlideChange는 Swiper에서 제공하는 prop이기 때문에 함께 분리하는 게 좋아보여요

```suggestion
      <Swiper
        className="swiper"
        {...swiperSettings}>
```

Reply:
그래서 `useRecoilState(sliderIdxState)` 까지 `useCardSwiper` 훅으로 분리함으로써

1. 컨포넌트 단에서는 UI 정보만 받아 UI를 보여주는 역할만 하게 되고,
2. 커스턴 훅 에서는 스와이퍼 정보들을 가공하고 `swiperSettings`만 내뱉는 역할을 해주는 거죠- 그렇게 되면 다른 컴포넌트에서도 재사용이 가능할 거구요

Reply:
아하 `useRecoilState(sliderIdxState)`까지 분리하는 방법이 있군요👀👀 해보겠습니다:)